stix_shifter_modules/reversinglabs/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,266 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+from stix_shifter_utils.stix_translation.src.utils.transformers import TimestampToMilliseconds
+from stix_shifter_utils.stix_translation.src.json_to_stix import observable
+import logging
+import re
+
+# Source and destination reference mapping for ip and mac addresses.
+# Change the keys to match the data source fields. The value array indicates the possible data type that can come into from field.
+REFERENCE_DATA_TYPES = {"SourceIpV4": ["ipv4", "ipv4_cidr"],
+                        "SourceIpV6": ["ipv6"],
+                        "DestinationIpV4": ["ipv4", "ipv4_cidr"],
+                        "DestinationIpV6": ["ipv6"]}
+
+logger = logging.getLogger(__name__)
+data = ""
+dataType = ""
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
```

Comment:
This shouldn't be needed as they are defined in the `operators.json` and brought in as part of the `BaseQueryTranslator` class. If you look at other connectors, you will see `self.comparator_lookup = self.dmm.map_comparator()` in the `QueryStringPatternTranslator` initialization.

Reply:
Updated